pkg/workloadmeta/cached_entity.go

```diff
@@ -0,0 +1,74 @@
+package workloadmeta
+
+import (
+	"sort"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type cachedEntity struct {
+	cached        Entity
+	sources       map[Source]Entity
+	sortedSources []string
+}
```

Comment:
is it okay that `cachedEntity` is not thread safe?

Reply:
Yes, they're protected by `Store.storeMut` in the store itself, and no references to it are returned to calling code. I added a comment clarifying.